internal/provider/kubernetes/gatewayclass.go

```diff
@@ -96,21 +102,27 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, request reconcil
 
 	var cc controlledClasses
 
+	found := false
 	for i := range gatewayClasses.Items {
 		if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
 			cc.addMatch(&gatewayClasses.Items[i])
+			if gatewayClasses.Items[i].GetName() == request.Name {
+				found = true
+			}
 		}
 	}
+	if !found {
+		r.resourceTable.GatewayClasses.Delete(request.Name)
+	}
 
 	// no controlled gatewayclasses, trigger a delete
 	if len(cc.matchedClasses) == 0 {
 		r.log.Info("failed to find gatewayclass", "name", request.Name)
```

Comment:
I think you want to issue a ResourceTable delete here per the comment -- this handles the case where the GatewayClass was deleted, so it can't be fetched from the API server anymore, so we delete it from the ResourceTable since we shouldn't be programming it anymore.